src/harvester/stats.rs

```diff
@@ -0,0 +1,129 @@
+use std::env::var;
+use std::sync::Mutex;
+use std::time::Duration;
+
+use anyhow::Result;
+use askama::Template;
+use lettre::{
+    message::{header::ContentType, Mailbox, SinglePart},
+    AsyncSmtpTransport, AsyncTransport, Message, Tokio1Executor,
+};
+
+#[derive(Default, Debug)]
+pub struct Stats(Mutex<Vec<StatsInner>>);
+
+impl Stats {
+    pub fn record_harvest(
+        &self,
+        source: String,
+        duration: Duration,
+        count: usize,
+        transmitted: usize,
+        failed: usize,
+    ) {
+        self.0.lock().unwrap().push(StatsInner {
+            name: source,
+            duration,
+            count,
+            transmitted,
+            failed,
+        });
+    }
+
+    pub async fn mail_summary(self, duration: Duration) -> Result<()> {
```

Comment:
The summary statistics should probably include the `start` in addition to the duration so that we know how current the data is (also per source), especially if this is exported via a `/metrics` route.